internal/commands/app/create_inputs_test.go

```diff
@@ -0,0 +1,272 @@
+package app
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/cloud/atlas"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/local"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/10gen/realm-cli/internal/utils/test/mock"
+
+	"github.com/Netflix/go-expect"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func TestAppCreateInputsResolve(t *testing.T) {
+	for _, tc := range []struct {
+		description string
+		inputs      createInputs
+		procedure   func(c *expect.Console)
+		test        func(t *testing.T, i createInputs)
+	}{
+		{
+			description: "with no flags set should prompt for just name and set realm.Location and deployment model to defaults",
+			procedure: func(c *expect.Console) {
+				c.ExpectString("App Name")
+				c.SendLine("test-app")
+				c.ExpectEOF()
+			},
+			test: func(t *testing.T, i createInputs) {
+				assert.Equal(t, "test-app", i.Name)
+				assert.Equal(t, flagDeploymentModelDefault, i.DeploymentModel)
+				assert.Equal(t, flagLocationDefault, i.Location)
+			},
+		},
+		{
+			description: "with a name flag set should prompt for nothing else and set realm.Location and deployment model to defaults",
+			inputs:      createInputs{newAppInputs: newAppInputs{Name: "test-app"}},
+			procedure:   func(c *expect.Console) {},
+			test: func(t *testing.T, i createInputs) {
+				assert.Equal(t, "test-app", i.Name)
+				assert.Equal(t, flagDeploymentModelDefault, i.DeploymentModel)
+				assert.Equal(t, flagLocationDefault, i.Location)
+			},
+		},
+		{
+			description: "with name realm.Location and deployment model flags set should prompt for nothing else",
+			inputs: createInputs{newAppInputs: newAppInputs{
+				Name:            "test-app",
+				DeploymentModel: realm.DeploymentModelLocal,
+				Location:        realm.LocationOregon,
+			}},
+			procedure: func(c *expect.Console) {},
+			test: func(t *testing.T, i createInputs) {
+				assert.Equal(t, "test-app", i.Name)
+				assert.Equal(t, realm.DeploymentModelLocal, i.DeploymentModel)
+				assert.Equal(t, realm.LocationOregon, i.Location)
+			},
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			profile := mock.NewProfile(t)
+
+			_, console, _, ui, consoleErr := mock.NewVT10XConsole()
+			assert.Nil(t, consoleErr)
+			defer console.Close()
+
+			doneCh := make(chan (struct{}))
+			go func() {
+				defer close(doneCh)
+				tc.procedure(console)
+			}()
+
+			assert.Nil(t, tc.inputs.Resolve(profile, ui))
+
+			console.Tty().Close() // flush the writers
+			<-doneCh              // wait for procedure to complete
+
+			tc.test(t, tc.inputs)
+		})
+	}
+}
+
+func TestAppCreateInputsResolveAppName(t *testing.T) {
+	testApp := realm.App{
+		ID:          primitive.NewObjectID().Hex(),
+		GroupID:     primitive.NewObjectID().Hex(),
+		ClientAppID: "test-app-abcde",
+		Name:        "test-app",
+	}
+
+	for _, tc := range []struct {
```

Comment:
this is a loaded test case `struct`... and im not sure we actually need the interactive console here, right?  (every `procedure` here is an empty func...)

I think the `"should return an error when client fails to find apps"` should be its own `t.Run` block.  you can table the other two together if you'd like, and the test case struct should end up being much simpler